### PR TITLE
Support m.login.token

### PR DESCRIPTION
**Pull request by erikjohnston**
Support `m.login.token` login flow.

https://github.com/matrix-org/matrix-doc/blob/master/specification/1-client_server_api.rst#token-based

### CODE REVIEW

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/LoginRestClient.java [hunk 1]
```diff
@@ -75,4 +78,47 @@ public void success(JsonObject jsonObject, Response response) {
             }
         });
     }
+
+    /**
+     * Attempt a user/password log in.
+     * @param user the user name
+     * @param token the password
```
Comment (Kegsay) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/LoginRestClient.java]: Is "password" really the right word here?

Comment (erikjohnston) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/LoginRestClient.java]: Probably not

Comment (erikjohnston) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/LoginRestClient.java]: done

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**